app/knights/main_stats.py

```diff
@@ -0,0 +1,19 @@
+def knight_stat(all_knights, knight_name):
+    knight = all_knights[knight_name]
+    # apply armour
+    knight["protection"] = 0
+    for part in knight["armour"]:
+        knight["protection"] += part["protection"]
+        # apply weapon
+    knight["power"] += knight["weapon"]["power"]
+    # apply potion if exist
+    if knight["potion"] is not None:
+        if "power" in knight["potion"]["effect"]:
+            knight["power"] += knight["potion"]["effect"]["power"]
+
+        if "protection" in knight["potion"]["effect"]:
+            knight["protection"] += knight["potion"]["effect"]["protection"]
+
+        if "hp" in knight["potion"]["effect"]:
+            knight["hp"] += knight["potion"]["effect"]["hp"]

```

Comment:
DRY, create a loop, and `stats = ("hp", "protection", "power")`  and looping on `stats`